sympy/functions/elementary/trigonometric.py

```diff
@@ -2279,8 +2285,26 @@ class asec(InverseTrigonometricFunction):
     Notes
     =====
 
-    asec(x) will evaluate automatically in the cases
-    oo, -oo, 0, 1, -1.
+    ``asec(x)`` will evaluate automatically in the cases
+    ``oo``, ``-oo``, ``0``, ``1``, ``-1``.
+
+    ``asec(x)`` has branch cut in the interval [-1, 1]. For complex arguments,
+    it can be defined [4]_ as
+
+    .. math::
+        sec^{-1}(z) = -i*(log(\sqrt{1 - z^2} + 1) - log(z))
+
+    At ``x = 0``, for positive branch cut, the limit evaluates to ``zoo``. For
+    negative branch cut, the limit
+
+    .. math::
+        \lim_{z \to 0}-i*(log(-\sqrt{1 - z^2} + 1) - log(z))
+
+    we get the derivative of :math:`-\sqrt(1 - z^2)` which ultimately evaluates
```

Comment:
Actually, we get (-i times) the *logarithm* of the derivative.

Perhaps we should write the definition in the form ``-i*\log((sqrt{1 - z^2} + 1)/z)`` and consider the limits of the two branches of the argument of the logarithm. It is not necessary to copy the exact form from [4].